src/ray/raylet/node_manager.cc

```diff
@@ -1096,16 +1096,23 @@ void NodeManager::HandleUnexpectedWorkerFailure(const rpc::WorkerDeltaData &data
       if (!worker_id.IsNil()) {
         // If the failed worker was a leased worker's owner, then kill the leased worker.
         if (owner_worker_id == worker_id) {
-          RAY_LOG(INFO) << "Owner process " << owner_worker_id
-                        << " died, killing leased worker " << worker->WorkerId();
-          KillWorker(worker);
+          std::ostringstream stream;
+          stream << "The leased worker " << worker->WorkerId()
+                 << " is killed because the owner process " << owner_worker_id
+                 << " died.";
+          const auto &err_msg = stream.str();
+          RAY_LOG(INFO) << err_msg;
+          DestroyWorker(worker, rpc::WorkerExitType::SYSTEM_ERROR, err_msg);
```

Comment:
This file change is to add more detail when a worker is returned with was_error = true. I found it currently doesn't give any detail